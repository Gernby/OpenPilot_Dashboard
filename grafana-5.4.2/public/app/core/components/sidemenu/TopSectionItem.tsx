import React, { SFC } from 'react';
import SideMenuDropDown from './SideMenuDropDown';

export interface Props {
  link: any;
}

const TopSectionItem: SFC<Props> = props => {
  const { link } = props;
  return (
    <div className="sidemenu-item dropdown">
      <a className="sidemenu-link" href={link.url} target={link.target}>
        <span className="icon-circle sidemenu-icon">
          <i className={link.icon} />
          {link.img && <img src={link.img} />}
        </span>
      </a>
      {link.children && <SideMenuDropDown link={link} />}
    </div>
  );
};

export default TopSectionItem;
