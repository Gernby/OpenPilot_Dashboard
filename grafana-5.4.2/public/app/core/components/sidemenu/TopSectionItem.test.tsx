import React from 'react';
import { shallow } from 'enzyme';
import TopSectionItem from './TopSectionItem';

const setup = (propOverrides?: object) => {
  const props = Object.assign(
    {
      link: {},
    },
    propOverrides
  );

  return shallow(<TopSectionItem {...props} />);
};

describe('Render', () => {
  it('should render component', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
