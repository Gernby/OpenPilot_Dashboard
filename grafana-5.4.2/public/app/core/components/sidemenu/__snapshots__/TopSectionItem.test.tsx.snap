// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<div
  className="sidemenu-item dropdown"
>
  <a
    className="sidemenu-link"
  >
    <span
      className="icon-circle sidemenu-icon"
    >
      <i />
    </span>
  </a>
</div>
`;
